src/RA_Dlg_AchievementsReporter.cpp

```diff
@@ -30,78 +27,86 @@ void Dlg_AchievementsReporter::SetupColumns(HWND hList)
     //	Remove all data.
     ListView_DeleteAllItems(hList);
 
-    LV_COLUMN col;
-    ZeroMemory(&col, sizeof(col));
-
-    for (size_t i = 0; i < SIZEOF_ARRAY(COL_TITLE); ++i)
+    auto i{ 0 };
+    for (const auto& sTitle : COL_TITLE)
     {
-        col.mask = LVCF_TEXT | LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT;
-        col.cx = COL_SIZE[i];
-        col.cchTextMax = 255;
-        ra::tstring str = NativeStr(COL_TITLE[i]);	//	Hold the temporary object
-        col.pszText = str.data();
-        col.iSubItem = i;
-
-        col.fmt = LVCFMT_LEFT | LVCFMT_FIXED_WIDTH;
-        if (i == SIZEOF_ARRAY(COL_TITLE) - 1)	//If the last element: fill to the end
+        ra::tstring str{ sTitle };
+        LV_COLUMN col
+        {
+            col.mask       = ra::to_unsigned(LVCF_TEXT | LVCF_WIDTH | LVCF_SUBITEM | LVCF_FMT),
+            col.fmt        = LVCFMT_LEFT | LVCFMT_FIXED_WIDTH,
+            col.cx         = COL_SIZE.at(i),
+            col.pszText    = str.data(),
+            col.cchTextMax = 255,
+            col.iSubItem   = i
+        };
+
+        if (i == ra::to_signed(COL_TITLE.size() - 1))	//If the last element: fill to the end
             col.fmt |= LVCFMT_FILL;
 
-        ListView_InsertColumn(hList, i, reinterpret_cast<LPARAM>(&col));
+        ListView_InsertColumn(hList, i, &col);
+        i++;
     }
 
     ms_nNumOccupiedRows = 0;
 }
 
-int Dlg_AchievementsReporter::AddAchievementToListBox(HWND hList, const Achievement* pAch)
+_Use_decl_annotations_
+int Dlg_AchievementsReporter::AddAchievementToListBox(HWND hList, const Achievement* const pAch)
 {
-    for (size_t i = 0; i < NumReporterColumns; ++i)
+    // We aren't actually using the value so we're using iterators
+    for (auto it = COL_TITLE.cbegin(); it != COL_TITLE.cend(); ++it)
     {
-        switch (i)
+        const auto nPos{ std::distance(COL_TITLE.cbegin(), it) };
+        switch (const auto col{ ra::itoe<Column>(nPos) }; col)
```

Comment:
Why do you need to define `col` here? Can't you just:
```
switch (ra::itoe<Column>(nPos))
```

Reply:
Yeah that can be done.